ILEditing/VanillaBugfixingILChanges.cs

```diff
@@ -69,5 +70,47 @@ private void FixHeldProjectileBlendState(On_PlayerDrawLayers.orig_DrawHeldProj o
             }
         }
         #endregion
+
+        #region Fix Vanilla Not Accounting For Multiple Bobbers When Fishing With Truffle Worm
+        private void FixTruffleWormFishing(ILContext il)
+        {
+            var cursor = new ILCursor(il);
+
+            // Initialize a flag variable whether truffle worm was used.
+            il.Method.Body.Variables.Add(new VariableDefinition(il.Module.TypeSystem.Boolean));
+            cursor.Emit(OpCodes.Ldc_I4_0);
+            cursor.Emit(OpCodes.Stloc_S, (byte)5);
+
+            // Find the call to Player.ItemCheck_CheckFishingBobber_PickAndConsumeBait.
+            if (!cursor.TryGotoNext(MoveType.Before, i => i.MatchCall<Player>("ItemCheck_CheckFishingBobber_PickAndConsumeBait")))
+            {
+                LogFailure("FixTruffleWormFishing", "Could not locate the call to Player.ItemCheck_CheckFishingBobber_PickAndConsumeBait.");
+                return;
+            }
+
+            // Skip if truffle worm was already used.
+            cursor.Index -= 4; // One should do this before arguments get pushed.
+            var label = il.DefineLabel();
+            cursor.Emit(OpCodes.Ldloc_S, (byte)5);
+            cursor.Emit(OpCodes.Brtrue_S, label);
+            cursor.Index += 4;
+
+            // Retrive baitTypeUsed, compare with truffle worm, and save it.
+            cursor.Index++;
```

Comment:
These are especially egregious.

Reply:
This and the first one could by amended by doing two MoveNext for `beq.s` and the function call.